types/src/blob/commitment.rs

```diff
@@ -25,40 +25,43 @@ impl Commitment {
     ///
     /// [Message layout rationale]: https://github.com/celestiaorg/celestia-specs/blob/e59efd63a2165866584833e91e1cb8a6ed8c8203/src/rationale/message_block_layout.md?plain=1#L12
     /// [Non-interactive default rules]: https://github.com/celestiaorg/celestia-specs/blob/e59efd63a2165866584833e91e1cb8a6ed8c8203/src/rationale/message_block_layout.md?plain=1#L36
-    pub fn generate(
+    pub fn for_blob(
         namespace: Namespace,
         share_version: u8,
         blob_data: &[u8],
     ) -> Result<Commitment> {
         let shares = split_blob_to_shares(namespace, share_version, blob_data)?;
+        Self::for_shares(namespace, &shares)
+    }
 
+    /// Generate the commitment for given shares.
+    pub fn for_shares(namespace: Namespace, mut shares: &[impl AsRef<[u8]>]) -> Result<Commitment> {
```

Comment:
question, should we require namespace there and check if all shares have this namespace inside?

Reply:
Another one, should I check if only the first one has sequence len and all are together taking a sequence len bytes +  optional padding?

Reply:
Would it be wrong to just have arbitrary bytes there and not verify them as shares?

Reply:
I added a comment that we treat the shares there as arbitrary byte blobs and perform no checks. Please resolve this issue if you agree